Modules/database_manager.py

```diff
@@ -0,0 +1,249 @@
+"""
+database_manager.py - Manages access to the central database file
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+"""
+import pyodbc
+from config import config
+import logging
+from functools import wraps
+from utils.ratlib import Singleton
+
+log = logging.getLogger(f"mecha.{__name__}")
+
+enabled = True
+
+
+def check_dbm_enabled(*args):
+    """
+    checks if the dbm-module is enabled or not.
+    """
+
+    def real_decorator(func):
+        """
+        The actual decorator
+
+        Args:
+            func (function): wrapped function
+
+        Returns:
+            function
+        """
+
+        @wraps(func)
+        async def wrapper():
+            """
+            Returns:
+                whatever the called function returns (probably None)
+            """
+            if not enabled:
+                raise ValueError("Module is disabled!")
+            return await func()
+        return wrapper
+    return real_decorator
+
+
+# noinspection SqlNoDataSourceInspection
+class DatabaseManager(metaclass=Singleton):
+
+    def __init__(self):
+        """
+        Connects to the DB, sets up the connection, retrieves the cursor.
+        Creates the default tables should they not exist.
+        """
+        # connect to PostgreSQL (PSQL) database
+        __config: dict = config.get("database")

```

Comment:
Imo ["stuff"] should always be reserved for collections, not complex classes/objects. This just gives you a clear distinction between what is what. 

Also this it no supposed to mangle the name. Using "config" as a name will cause name shadowing, which you should prevent if not completely necessary. I just chose __ as a prefix to denote the fact that this is a completely and utterly internal variable.

Reply:
using `get` is an unnecessary function call and does lead to confusion. usage of indexing is much clearer here.

Reply:
It may be an unnecessary function call but it emphasizes that this is *not* a collection, which is especially important in not strongly-typed languages like python.

Reply:
To quote from the docstring provided for `dict.get`
```Return the value for key if key is in the dictionary, else default. If default is not given, it defaults to None, so that this method never raises a KeyError.```

The only difference between using the function and using the index is that the function will return None if the key does not exist. I do not see where this function in any way implies the type returned. Further, the `database` key *is* a collection of subkeys.

Reply:
My argumentation always referred to implications about the object .get is executed on, not the returned type

Reply:
@beepbeat, our config **is** just a dict, so by your own logic you would use indexing.

`get` returning None if nothing is found is also something we don't want. When something is on fire, we want to know about it as soon as possible, i.e. we want it to raise an exception.

The Python convention for avoiding name shadowing or keywords is generally to append a `_` or, in this case, you could simply to rename the variable to better reflect its relationship to the one it shadows. 
Prefixing names with underscores has special uses, please don't make up your own.

Reply:
If it's a dict, I'll gladly change it to use indexing, for me the config was always a class.

Due to how it's used, it'll raise an exception a few lines later, as None does not expose any members. 

If that's a convention in python to append, I'm sorry I haven't read through all the 420+ PEPs. I'm used to prepend from my previous style convention I followed.

Reply:
@beepbeat, neither have I and it's alright to not know the convention, that is why I am telling it to you.
(on a side note, the conventions are only in pep8, most peps are interpreter or standard library features.)